### PR TITLE
[google_sign_in] Bump GMS dependency and remove the need for Jetifier

### PR DESCRIPTION
**Pull request by ened**
Bump to latest GMS Auth version and replace the `guava` dependency with `guava-android`.
Also disables jetifier in the example project to verify.

Fixes https://github.com/flutter/flutter/issues/96361

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

### CODE REVIEW

File: packages/google_sign_in/google_sign_in/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ## NEXT
```
Comment (stuartmorgan) [inline packages/google_sign_in/google_sign_in/CHANGELOG.md:1]: This is a client-affecting change, and this needs a version bump

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
I don't have any context on the substantive changes here; @blasten should review this.

File: packages/google_sign_in/google_sign_in/example/android/gradle.properties [hunk 2]
```diff
@@ -1,4 +1,4 @@
 org.gradle.jvmargs=-Xmx1536M
 android.enableR8=true
 android.useAndroidX=true
-android.enableJetifier=true
+android.enableJetifier=false
```
Comment (blasten) [inline packages/google_sign_in/google_sign_in/example/android/gradle.properties:4]: this line can be deleted since enableJetifier is false by default

**Review by blasten — COMMENTED** [review]
lgtm + two comments

File: packages/google_sign_in/google_sign_in/CHANGELOG.md (hunk 1 above)
Comment (ened) [reply packages/google_sign_in/google_sign_in/CHANGELOG.md]: ok , 5.2.2

File: packages/google_sign_in/google_sign_in/example/android/gradle.properties (hunk 2 above)
Comment (ened) [reply packages/google_sign_in/google_sign_in/example/android/gradle.properties]: removed

File: packages/google_sign_in/google_sign_in/CHANGELOG.md [hunk 3]
```diff
@@ -1,7 +1,8 @@
 ## 5.2.2
 
 * Updates Android compileSdkVersion to 31.
-* Removes depenedncy on `meta`.
+* Bump Android dependency on com.google.android.gms:play-services-auth and remove the need for jetifier.
```
Comment (stuartmorgan) [inline packages/google_sign_in/google_sign_in/CHANGELOG.md:4]: Looks like you merged with another 5.2.2 bump; you'll need to move this to 5.2.3.

Comment (ened) [reply packages/google_sign_in/google_sign_in/CHANGELOG.md]: done.

File: packages/google_sign_in/google_sign_in/CHANGELOG.md [hunk 4]
```diff
@@ -1,7 +1,10 @@
+## 5.2.3
+* Bump Android dependency on com.google.android.gms:play-services-auth and remove the need for jetifier.
```
Comment (stuartmorgan) [inline packages/google_sign_in/google_sign_in/CHANGELOG.md:2]: Bumps

(See discussion of verb form in the new CHANGELOG style guide linked from the PR checklist)

File: packages/google_sign_in/google_sign_in/CHANGELOG.md [hunk 5]
```diff
@@ -1,7 +1,10 @@
+## 5.2.3
```
Comment (stuartmorgan) [inline packages/google_sign_in/google_sign_in/CHANGELOG.md:1]: Blank line between heading and list to match all the other entries please.

**Review by stuartmorgan — APPROVED** [review]
Metadata LGTM with nits.

File: packages/google_sign_in/google_sign_in/CHANGELOG.md (hunk 5 above)
Comment (ened) [reply packages/google_sign_in/google_sign_in/CHANGELOG.md:1]: done

File: packages/google_sign_in/google_sign_in/CHANGELOG.md (hunk 4 above)
Comment (ened) [reply packages/google_sign_in/google_sign_in/CHANGELOG.md:2]: done

**Comment by ened** [comment]
@blasten Could you kindly approve so that I can merge?

**Review by blasten — APPROVED** [review]

**Pull request merged**